src/main/pages/setup-docker/docker-cli.adoc

```diff
@@ -8,142 +8,216 @@ folder: setup-docker
+|Starts test on the Che instance
+
+|`upgrade`                             
+|Upgrades Che from one version to another with migrations and backups
+
+|`version`                             
+|Installed version and upgrade paths
+|===
+
+The following table lists the global command options that you can se with the Docker CLI image.
+
+[cols="2*", options="header"]
+|===
+|Command option
+|Description
+|`--fast`                               
+|Skips networking, version, nightly, and preflight checks
+
+|`--offline`                            
+|Runs CLI in the offline mode, loading images from the disk
+
+|`--debug`                              
+|Enable debugging of the Che server
+
+|`--trace`                              
+|Activates trace output for debugging CLI
+|===
 
-Generates a Che instance configuration thta is placed in `/instance`. This command uses puppet to generate Docker Compose configuration files to run Che and its associated server. Che’s server configuration is generated as a che.properties file that is volume mounted into the Che server when it boots. This command is executed on every `start` or `restart`.
+The CLI has the following three primary phases:
 
-If you are using a `eclipse/che:<version>` image and it does not match the version that is in `/instance/che.ver`, then the configuration will abort to prevent you from running a configuration for a different version than what is currently installed.
+* The initialization phase is executed by the `init` command. It installs version-specific files into the directory mounted to `/data`. This includes the universal configuration file named `che.env`, a version identifier, and a location where configuration files are saved. The configuration is executed by the `config` command. It takes as input, the `che.env` configuration file, the OS of your host, and then generates an OS-specific set of configuration files in the `/data/instance` directory that can be used to run an instance of Che. 
 
-This command respects `--no-force`, `--pull`, `--force`, and `--offline`.
+* The configuration phase runs an initialization if a directory is not found. Every execution of the `config` command overwrites the files in the `/data/instance` directory with the latest configuration. This ensures that if an administrator modifies any configuration file, the instance’s configuration files are updated consistently. The CLI generates a large number of configuration files specific to running Che. The configuration files are sourced from the puppet templates that are stored in the GitHub repository under `/dockerfiles/init`. 
 
-'''''
+* The start phase is executed by the `start` command and it uses a configuration-generated `docker-compose-container.yml` file to launch Eclipse Che. The start phase always executes a `config` command, so any files that were edited in the `/data/instance` directory are overwritten with the generated configuration from the CLI.
```

Comment:
"The start phase is executed" and "it uses ... to launch Eclipse Che" are slightly different ideas, so I'd place them in separate sentences.